File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int ans = 0;
        // Count frequency of each number
        for(int num : nums){
            map.put(num, map.getOrDefault(num,0)+1);
        }
        // Check for each number if num+1 exists
        for(int key : map.keySet()){
            if(map.containsKey(key+1)){
                ans = Math.max(ans, map.get(key) + map.get(key+1));
            }
        }
        return ans;
    }
}

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> map;
        int maxLen = 0;

        // Count frequency of each number
        for (int num : nums) {
            map[num]++;
        }

        // Check for each number if num + 1 exists
        for (auto& pair : map) {
            int key = pair.first;
            if (map.count(key + 1)) {
                maxLen = max(maxLen, map[key] + map[key + 1]);
            }
        }

        return maxLen;
    }
};